### PR TITLE
[Impeller] Improved documentation of the gaussian blur.

### PR DESCRIPTION
**Pull request by gaaclarke**
Just trying to capture some of my thoughts as I look into some blur bugs.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]

File: impeller/entity/contents/filters/filter_contents.cc [hunk 1]
```diff
@@ -55,11 +55,22 @@ std::shared_ptr<FilterContents> FilterContents::MakeGaussianBlur(
     Sigma sigma_y,
     BlurStyle blur_style,
     Entity::TileMode tile_mode) {
-  auto x_blur = MakeDirectionalGaussianBlur(
-      input, sigma_x, Point(1, 0), BlurStyle::kNormal, tile_mode, false, {});
-  auto y_blur = MakeDirectionalGaussianBlur(FilterInput::Make(x_blur), sigma_y,
-                                            Point(0, 1), blur_style, tile_mode,
-                                            true, sigma_x);
+  std::shared_ptr<FilterContents> x_blur = MakeDirectionalGaussianBlur(
+      /*input=*/input,
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply impeller/entity/contents/filters/filter_contents.cc:59]: The big thing for me was being able to see what `true` and `{}` is.  There is the clang_tidy check `bugprone-argument-comment` which I swear I turned on at some point.  That makes sure the comments match the parameters.

I've never seen an IDE plugin that adds the comments.  It should be possible with vscode.

[the remark this answers is not in the document]
Comment (chinmaygarde) [reply impeller/entity/contents/filters/filter_contents.cc:59]: > I've never seen an IDE plugin that adds the comments. It should be possible with vscode.

Yeah, it does seem like toil that should been automated away by this point.

File: impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+///  |             |        |     |
+///  |             |        |     |        Pass 2
+///  |             |        |     |        +----+
+///  |             |        |     |        |    |
+///  |             |  ->    |     |  ->    |    |
+///  |             |        |     |        |    |
+///  |             |        |     |        +----+
+///  |             |        |     |        87x102
+///  +-------------+        +-----+
+///     586x678             97x678
```
Comment (bdero) [inline impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.h:34]: Haha, nice. Love this graphic. :)

**Review by bdero — APPROVED** [review]
LGTM, thanks!

File: impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.h (hunk 2 above)
Comment (gaaclarke) [reply impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.h:34]: dude, emacs "artist mode"

Comment (chinmaygarde) [reply impeller/entity/contents/filters/directional_gaussian_blur_filter_contents.h:34]: https://monodraw.helftone.com/

**Pull request merged**